src/components/create-safe/status/usePendingSafeCreation.ts

```diff
@@ -19,18 +24,42 @@ export const pollSafeInfo = async (chainId: string, safeAddress: string): Promis
   })
 }
 
-export const checkSafeCreationTx = async (provider: JsonRpcProvider, txHash: string) => {
+export const checkSafeCreationTx = async (provider: JsonRpcProvider, txHash: string, chainId: string) => {
   const TIMEOUT_TIME = 6.5
 
   try {
-    const receipt = await provider.waitForTransaction(txHash, 1, TIMEOUT_TIME * 60_000)
+    const blockNumber = await provider.getBlockNumber()
+    const txResponse = await provider.getTransaction(txHash)
+    const proxyContractAddress = getProxyFactoryContractInstance(chainId).address
+
+    const replacement = {
+      data: txResponse.data,
+      from: txResponse.from,
+      nonce: txResponse.nonce,
+      to: proxyContractAddress,
+      value: txResponse.value,
+      startBlock: txResponse.blockNumber || blockNumber,
```

Comment:
If a transaction already exists, we are still requesting the current block number as well.

```suggestion
    const txResponse = await provider.getTransaction(txHash)
    const proxyContractAddress = getProxyFactoryContractInstance(chainId).address

    const replacement = {
      data: txResponse.data,
      from: txResponse.from,
      nonce: txResponse.nonce,
      to: proxyContractAddress,
      value: txResponse.value,
      startBlock: txResponse.blockNumber || await provider.getBlockNumber(),
```

Reply:
Nice!